From: PupJune2 <PupJune2@users.noreply.github.com>
Date: Sat, 17 Oct 2026 11:00:00 +0000
Subject: CromIKO New Tab Page

- Show the CromIKO logo (mascot and wordmark) above the search box on
  the new tab page, for every search engine. Doodles and search engine
  logos are no longer downloaded. In dark mode and on custom
  backgrounds the logo is tinted like the Google logo was.
- Fix the Bookmarks and Downloads shortcut icons, which used a light
  grey meant for dark backgrounds and were barely visible in light
  mode; they now use the theme accent color.

The logo is generated by tools/cromiko/gen_drawables.py.

License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
---
 .../android/java/res/layout/ntp_shortcuts.xml |  4 +-
 .../logo/java/res/drawable/ic_google_logo.xml | 71 ++++++++++++-------
 .../chrome/browser/logo/LogoMediator.java     | 12 +++-
 .../search_engines/TemplateUrlService.java    |  4 +-
 4 files changed, 62 insertions(+), 29 deletions(-)

diff --git a/chrome/android/java/res/layout/ntp_shortcuts.xml b/chrome/android/java/res/layout/ntp_shortcuts.xml
--- a/chrome/android/java/res/layout/ntp_shortcuts.xml
+++ b/chrome/android/java/res/layout/ntp_shortcuts.xml
@@ -36,7 +36,7 @@
             android:textAppearance="@style/TextAppearance.TextSmall.Secondary"
             app:drawableHeight="20sp"
             app:drawableWidth="20sp"
-            app:chromeDrawableTint="@color/default_icon_color_light"  />
+            app:chromeDrawableTint="@color/default_icon_color_accent1_tint_list"  />
     </FrameLayout>
 
     <FrameLayout
@@ -60,6 +60,6 @@
             android:textAppearance="@style/TextAppearance.TextSmall.Secondary"
             app:drawableHeight="20sp"
             app:drawableWidth="20sp"
-            app:chromeDrawableTint="@color/default_icon_color_light"  />
+            app:chromeDrawableTint="@color/default_icon_color_accent1_tint_list"  />
     </FrameLayout>
 </LinearLayout>
diff --git a/chrome/browser/ui/android/logo/java/res/drawable/ic_google_logo.xml b/chrome/browser/ui/android/logo/java/res/drawable/ic_google_logo.xml
--- a/chrome/browser/ui/android/logo/java/res/drawable/ic_google_logo.xml
+++ b/chrome/browser/ui/android/logo/java/res/drawable/ic_google_logo.xml
@@ -3,30 +3,53 @@ Copyright 2026 The Chromium Authors
 Use of this source code is governed by a BSD-style license that can be
 found in the LICENSE file.
 -->
-
+<!-- CromIKO: the CromIKO logo replaces the Google logo on the new tab page.
+     Generated by tools/cromiko/gen_drawables.py from tools/cromiko/cromiko-logo.svg. -->
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:width="270dp"
+    android:width="281dp"
     android:height="82dp"
-    android:viewportWidth="270"
-    android:viewportHeight="82"
-    tools:ignore="VectorRaster">
-  <path
-      android:pathData="M137.63,63.45C133.28,63.45 129.36,62.45 125.89,60.44C122.44,58.44 119.74,55.68 117.79,52.18C115.87,48.65 114.91,44.67 114.91,40.27C114.91,35.75 115.9,31.75 117.88,28.27C119.88,24.77 122.61,22.05 126.06,20.13C129.5,18.21 133.36,17.25 137.63,17.25C141.95,17.25 145.84,18.23 149.28,20.18C152.76,22.13 155.49,24.85 157.46,28.36C159.44,31.86 160.43,35.83 160.43,40.27C160.43,44.67 159.46,48.65 157.51,52.18C155.56,55.71 152.84,58.48 149.37,60.48C145.92,62.46 142.01,63.45 137.63,63.45ZM137.63,55.4C140.31,55.4 142.69,54.73 144.75,53.41C146.84,52.05 148.45,50.23 149.58,47.94C150.74,45.62 151.32,43.06 151.32,40.27C151.32,37.5 150.74,34.98 149.58,32.72C148.45,30.43 146.84,28.62 144.75,27.3C142.66,25.97 140.28,25.3 137.63,25.3C135.14,25.3 132.85,25.94 130.76,27.21C128.67,28.48 127,30.26 125.76,32.55C124.54,34.81 123.94,37.38 123.94,40.27C123.94,43.04 124.5,45.58 125.63,47.9C126.79,50.19 128.4,52.01 130.46,53.36C132.56,54.72 134.94,55.4 137.63,55.4Z"
-      android:fillColor="@color/google_logo_light_yellow"/>
-  <path
-      android:pathData="M15.43,59.13C20.23,61.95 25.54,63.4 31.37,63.45C37.24,63.51 42.43,62.29 46.92,59.81C51.44,57.32 54.93,53.8 57.39,49.25C59.85,44.67 61.08,39.42 61.08,33.49C61.08,31.96 60.94,30.26 60.65,28.4H31.24V37.17H51.88C51.51,40.93 50.43,44.12 48.62,46.75C46.81,49.38 44.43,51.36 41.5,52.69C38.59,53.99 35.28,54.64 31.58,54.64C27.42,54.64 23.68,53.66 20.35,51.71C17.01,49.76 14.38,47.05 12.46,43.57C10.54,40.07 9.58,36.1 9.58,31.66C9.58,27.11 10.54,23.11 12.46,19.67C14.41,16.19 17.05,13.52 20.39,11.66C23.72,9.79 27.45,8.86 31.58,8.86C34.83,8.86 37.71,9.4 40.22,10.47C42.77,11.54 45.24,13.31 47.64,15.77L53.79,9.32C51.1,6.16 47.85,3.81 44.04,2.29C40.22,0.76 36.03,0 31.45,0C25.57,0 20.23,1.38 15.43,4.15C10.62,6.89 6.85,10.67 4.11,15.47C1.37,20.27 0,25.63 0,31.54C0,37.36 1.37,42.7 4.11,47.56C6.88,52.42 10.65,56.27 15.43,59.13Z"
-      android:fillColor="@color/google_logo_light_blue"/>
-  <path
-      android:pathData="M88.33,17.25C92.66,17.25 96.54,18.23 99.99,20.18C103.46,22.13 106.19,24.85 108.17,28.36C110.15,31.86 111.14,35.83 111.14,40.27C111.14,44.67 110.16,48.65 108.21,52.18C106.26,55.71 103.55,58.48 100.07,60.49C96.63,62.46 92.71,63.45 88.33,63.45C83.98,63.45 80.07,62.45 76.59,60.44C73.14,58.44 70.45,55.68 68.5,52.18C66.57,48.65 65.61,44.67 65.61,40.27C65.61,35.75 66.6,31.75 68.58,28.27C70.59,24.77 73.31,22.06 76.76,20.13C80.21,18.21 84.07,17.25 88.33,17.25ZM88.33,25.3C85.85,25.3 83.56,25.94 81.47,27.21C79.38,28.48 77.71,30.26 76.47,32.55C75.25,34.81 74.64,37.38 74.64,40.27C74.64,43.04 75.21,45.58 76.34,47.9C77.5,50.19 79.11,52.01 81.17,53.36C83.26,54.72 85.65,55.4 88.33,55.4C91.02,55.4 93.39,54.73 95.45,53.41C97.54,52.05 99.16,50.23 100.29,47.94C101.44,45.62 102.02,43.06 102.02,40.27C102.02,37.5 101.44,34.98 100.29,32.72C99.16,30.43 97.54,28.63 95.45,27.3C93.36,25.97 90.99,25.3 88.33,25.3Z"
-      android:fillColor="@color/google_logo_light_red"/>
-  <path
-      android:pathData="M184.34,17.51C187.28,17.51 189.98,18.09 192.43,19.24C194.89,20.4 196.81,22.06 198.2,24.2H198.37V18.82H207.18V57.05C207.18,73.5 200.01,81.72 185.65,81.72C180.79,81.72 176.78,80.7 173.61,78.67C170.48,76.63 167.86,73.4 165.77,68.96L173.74,65.44C175.07,67.96 176.68,69.92 178.57,71.34C180.47,72.78 182.78,73.5 185.52,73.5C189.71,73.5 192.88,72.28 195.06,69.85C197.26,67.42 198.37,64.03 198.37,59.68V55.74H198.07C197.39,56.76 196.43,57.79 195.19,58.83C193.95,59.85 192.42,60.73 190.61,61.46C188.8,62.17 186.82,62.52 184.68,62.52C180.41,62.52 176.71,61.57 173.57,59.68C170.43,57.79 168.03,55.16 166.37,51.8C164.7,48.41 163.87,44.52 163.87,40.14C163.87,35.82 164.7,31.94 166.37,28.53C168.03,25.08 170.41,22.38 173.49,20.43C176.6,18.48 180.21,17.51 184.34,17.51ZM185.69,25.73C183.18,25.73 180.96,26.32 179.04,27.51C177.12,28.67 175.62,30.32 174.55,32.47C173.47,34.62 172.93,37.09 172.93,39.89C172.93,42.82 173.47,45.37 174.55,47.52C175.65,49.66 177.16,51.3 179.08,52.43C181,53.56 183.19,54.13 185.65,54.13C188.25,54.13 190.51,53.52 192.43,52.31C194.35,51.09 195.82,49.41 196.84,47.26C197.86,45.09 198.37,42.61 198.37,39.84C198.37,37.02 197.86,34.55 196.84,32.43C195.82,30.28 194.35,28.62 192.43,27.47C190.51,26.31 188.26,25.73 185.69,25.73Z"
-      android:fillColor="@color/google_logo_light_blue"/>
-  <path
-      android:pathData="M214.01,1.36V62.05H223.25V1.36H214.01Z"
-      android:fillColor="@color/google_logo_light_green"/>
-  <path
-      android:pathData="M250.54,63.38C244.05,63.38 238.66,61.29 234.4,57.11C230.16,52.93 228.04,47.42 228.04,40.58C228.04,33.88 230.09,28.37 234.18,24.05C238.31,19.69 243.69,17.52 250.33,17.52C254.68,17.52 258.44,18.64 261.61,20.87C264.8,23.1 267.32,26.59 269.15,31.34C269.32,31.82 269.48,32.31 269.62,32.82C269.76,33.33 269.89,33.91 270,34.56L235.24,49.31L232.83,42.31L261.61,30.11L260.8,33.96C259.73,30.94 258.23,28.75 256.31,27.4C254.42,26.04 252.31,25.36 249.99,25.36C246.04,25.36 242.89,26.72 240.54,29.43C238.2,32.11 237.02,35.65 237.02,40.03C237.02,44.15 238.38,47.67 241.09,50.58C243.8,53.49 247.1,54.95 250.97,54.95C253.29,54.95 255.45,54.37 257.45,53.21C259.46,52.05 261.21,50.42 262.71,48.33L269.58,52.83C267.65,55.99 265.03,58.55 261.69,60.5C258.36,62.42 254.64,63.38 250.54,63.38Z"
-      android:fillColor="@color/google_logo_light_red"/>
+    android:viewportWidth="281"
+    android:viewportHeight="82">
+    <group
+        android:translateX="-76.4006"
+        android:translateY="-98.5249"
+        android:scaleX="0.1695"
+        android:scaleY="0.1695">
+        <path
+            android:fillType="evenOdd"
+            android:pathData="M803.93 593.11 C838.39 589.85 851.01 624.09 864.97 649.82 C884.68 685.41 902.32 722.11 917.8 759.73 C925.28 777.9 931.31 796.56 938.78 814.54 C936.01 817.99 931.67 822.69 928.52 825.7 C839.78 910.44 685.13 973.41 563.18 972.11 C534.86 971.05 495.27 965.19 475.34 943.51 C466.31 933.8 461.67 920.82 462.5 907.59 C464.55 870.78 500.89 834.18 527.1 811.09 C525.11 774.06 520.67 696.72 539.67 664.53 C564.4 622.62 616.8 661.34 645.13 679.34 C677.99 666.89 709.82 656.58 744.25 649.57 C762.72 629.72 777.86 599.34 803.93 593.11 Z M797.41 727.23 C848.2 720.94 860.38 755.72 871.47 795.57 C878.96 822.49 871.64 848.19 847.4 863.79 C833.33 872.84 807.12 879.43 791.41 884.51 C762.85 893.98 734.15 903.01 705.3 911.59 C669.3 919.28 641.37 902.72 633.44 866.84 C627.32 839.19 626.05 816.23 641.11 790.84 C655.96 765.79 679.85 762.23 705.24 753.6 C735.79 743.44 766.03 734.47 797.41 727.23 Z M814.02 766.18 C830.19 766.36 833.9 783.97 838.69 796.38 C844.23 810.73 847.51 825.01 830.19 831.19 C818.34 830.43 813.9 824.25 810.13 813.43 C804.41 796.99 791.12 774.86 814.02 766.18 Z M721.03 796.19 C729.88 795.12 738.03 798.5 741.22 807.04 C747.4 823.57 763.84 848.59 742.5 859.49 C721.79 862.61 721.57 841.99 715.01 827.2 C709.12 813.9 707.11 804.88 721.03 796.19 Z M528.61 833.78 C529.04 834.23 538.96 885.73 541.55 893.73 C547.09 910.85 552.48 923.54 560.45 940.09 C550 939.07 542.66 937.76 532.54 935.42 C473.97 915.88 495.05 867.32 528.61 833.78 Z M740.42 672.1 C742.82 671.69 746.1 671.21 748.47 672.04 C750.81 672.88 752.7 674.67 753.68 676.96 C754.66 679.26 754.61 681.62 753.48 683.87 C748.49 693.82 699.81 703.99 687.43 707.86 C684.67 708.42 681.73 708.54 679.11 707.39 C676.57 706.27 674.68 704.21 673.86 701.56 C673.26 699.65 673.46 697.65 674.47 695.89 C680.09 686.09 727.36 675.82 740.42 672.1 Z"
+            android:fillColor="#1E2024"/>
+        <path
+            android:pathData="M928.66 702.48 C929.31 702.53 929.97 702.58 930.62 702.65 C958.31 705.41 988.37 713.72 1006.68 735.9 C1046.14 790.42 989.39 843.43 953.83 879.86 C956.98 892.57 961.98 909.37 961.5 922.2 C960.32 953.38 939.16 964.4 911.52 963.76 C909.23 972.07 906.62 980.28 903.7 988.39 C892.25 1019.81 866.77 1063.68 823.8 1050.55 C809.47 1046.17 802.55 1025.85 797.09 1012.93 C736.01 1060.33 611.55 1072.24 561.65 1004.06 C572.23 1005.29 591.03 1004.35 601.99 1003.17 C695.26 993.1 796.68 951.13 873.87 898.75 C914.18 869.7 969.81 827.4 982.11 776.91 C990.37 743.06 957.07 726.34 928.82 721.65 L928.66 702.48 Z"
+            android:fillColor="#1E2024"/>
+        <path
+            android:pathData="M1006.68 735.9 C1046.14 790.42 989.39 843.43 953.83 879.86 L938.4 892.46 C934.96 888.54 931.41 884.15 928.04 880.12 C966.17 847.37 1027.42 791.53 1006.68 735.9 Z"
+            android:fillColor="#5C7CFC"/>
+        <path
+            android:pathData="M928 605.54 C929.99 607.52 936.6 627.46 939.22 631.81 C946.04 643.17 964.61 647.94 976.55 651.07 C943.29 660.87 938.05 665.19 928.72 698.31 C915.91 663.98 917.43 661.01 879.8 651.48 C912.83 641.16 918.04 638.91 928 605.54 Z"
+            android:fillColor="#5C7CFC"/>
+        <path
+            android:pathData="M1126.72 792.41 C1150.98 790.13 1182.21 793.26 1207.02 792.43 C1233.4 791.56 1207.42 819.5 1203.63 829.19 C1202.37 832.4 1198.25 837.61 1194.24 838.04 C1174.11 840.6 1151.03 836.4 1131.08 838.4 C1119.37 839.47 1108.61 845.3 1101.32 854.52 C1094.28 863.36 1091.08 874.66 1092.46 885.89 C1093.38 893.61 1096.56 900.9 1101.62 906.82 C1127.31 937.41 1186.25 910.94 1204.11 925.41 C1204.35 925.59 1204.57 925.79 1204.8 925.98 C1207.1 932.18 1209.64 963.76 1200.87 966.56 C1189.51 969.62 1176.3 967.65 1164.47 967.92 C1130.55 968.71 1100.01 969.53 1072.37 946.48 C1054.69 931.91 1043.66 910.81 1041.78 887.97 C1039.72 865.18 1046.87 842.51 1061.64 825.03 C1078.44 805.26 1101.12 794.63 1126.72 792.41 Z"
+            android:fillColor="#1E2024"/>
+        <path
+            android:pathData="M1272.79 826.32 C1283.83 825.44 1319.59 821.54 1321.64 835.15 C1331.01 897.49 1285.58 855.88 1271.28 878.62 C1263.5 890.98 1275.66 951.18 1265.24 965.91 L1264.9 966.02 C1256.68 968.79 1217.81 971.5 1218.69 957.32 C1220.16 933.66 1215.26 896.9 1219.75 874.48 C1221.45 865.88 1224.98 857.76 1230.1 850.65 C1240.8 835.77 1255.18 829.14 1272.79 826.32 Z"
+            android:fillColor="#1E2024"/>
+        <path
+            android:fillType="evenOdd"
+            android:pathData="M1382.66 823.72 C1422.77 816.88 1460.79 843.98 1467.43 884.12 C1474.06 924.27 1446.78 962.16 1406.6 968.59 C1366.7 974.97 1329.15 947.92 1322.56 908.06 C1315.98 868.2 1342.83 830.51 1382.66 823.72 Z M1390.38 871.42 C1403.64 868.94 1416.44 877.52 1419.19 890.72 C1421.95 903.92 1413.64 916.91 1400.5 919.94 C1391.75 921.96 1382.59 919.09 1376.56 912.43 C1370.53 905.78 1368.57 896.38 1371.43 887.87 C1374.3 879.36 1381.55 873.06 1390.38 871.42 Z"
+            android:fillColor="#1E2024"/>
+        <path
+            android:pathData="M1609.56 822.39 C1622.41 822.19 1634.57 824.26 1645.71 831.02 C1659.72 839.53 1668.95 853.7 1672.57 869.54 C1674.88 879.63 1676.44 956.22 1672.29 962.16 C1670.51 964.69 1668.05 966.3 1664.99 966.83 C1658.01 968.05 1633.1 968.68 1627.72 964.64 C1618.91 958.02 1625.54 898.11 1623.59 884.82 C1623.03 881.02 1621.5 876.84 1618.17 874.62 C1615.63 872.93 1612.05 872.58 1609.14 873.4 C1591.35 878.43 1609.41 948.12 1599.03 962.96 C1597.03 965.82 1593.68 966.65 1590.41 967.12 C1583.29 968.15 1559.48 969.14 1554.12 964.42 C1551.07 961.74 1550.93 958.03 1550.76 954.25 C1550.71 933.99 1550.67 913.72 1550.6 893.47 C1550.57 886.28 1551.8 877.49 1543.93 873.9 C1541.19 872.66 1538.07 872.62 1535.3 873.8 C1518.7 880.93 1534.61 944.24 1526.36 961.44 C1525.11 964.04 1523.16 965.4 1520.44 966.29 C1514.71 968.16 1487.61 968.43 1482.67 965.52 C1480.16 964.03 1479.14 961.49 1478.49 958.75 C1476.42 950.04 1477.17 882.5 1478.72 872.36 C1479.82 864.91 1482.35 857.73 1486.18 851.25 C1504.15 821.04 1544.97 814.5 1574.07 832.23 C1587.86 825.45 1594.3 823.58 1609.56 822.39 Z"
+            android:fillColor="#1E2024"/>
+        <path
+            android:pathData="M1692.72 784.18 C1703.07 783.46 1726.83 781.79 1734.7 788.3 C1738.58 799.84 1735.79 931.4 1736.73 954.02 C1737 960.62 1735.47 962.66 1731.02 966.89 C1721.67 967.75 1694.16 969.63 1687.76 963.03 C1686.69 958.74 1685.92 953.17 1685.95 948.74 C1686.21 896.37 1685.17 843.85 1686.19 791.53 C1686.25 788.31 1690.36 785.76 1692.72 784.18 Z"
+            android:fillColor="#1E2024"/>
+        <path
+            android:pathData="M1865.71 784.16 C1874.26 783.52 1913.52 782.39 1919.68 784.91 C1921.16 785.51 1922.72 786.76 1923.19 788.31 C1927.85 803.72 1851.31 856.43 1843.34 874.79 C1843.37 875.63 1843.41 876.47 1843.44 877.32 C1855.93 890.83 1882.65 914.52 1895.3 926.28 C1947.43 970.4 1928.16 970.49 1869.02 966.78 C1862.38 966.36 1809.83 914.55 1799.28 904.14 C1798.9 917.2 1799.39 932.49 1799.59 945.72 C1800.02 973.24 1786.96 967.48 1766.22 967.86 C1752.86 967.75 1748.35 966.53 1748.26 952.47 C1747.9 899.29 1748.06 846.07 1748.25 792.89 C1748.26 788.68 1751.76 786.57 1754.81 784.17 C1761.97 783.74 1770.34 783.82 1777.25 783.59 C1804.35 782.68 1799.63 797.26 1799.14 817.31 C1798.92 826.21 1799.04 837.38 1799.11 846.45 C1809.92 835.92 1856.34 789.75 1865.71 784.16 Z"
+            android:fillColor="#1E2024"/>
+        <path
+            android:fillType="evenOdd"
+            android:pathData="M1991.96 775.4 C2045.32 770.2 2092.83 809.12 2098.25 862.46 C2103.67 915.79 2064.95 963.47 2011.64 969.11 C1958.01 974.78 1909.99 935.78 1904.54 882.13 C1899.09 828.49 1938.29 780.63 1991.96 775.4 Z M1995.35 826.28 C2020.95 822.99 2044.32 841.19 2047.43 866.81 C2050.53 892.43 2032.17 915.68 2006.53 918.6 C1981.14 921.5 1958.17 903.36 1955.1 877.99 C1952.03 852.62 1970 829.53 1995.35 826.28 Z"
+            android:fillColor="#1E2024"/>
+    </group>
 </vector>
diff --git a/chrome/browser/ui/android/logo/java/src/org/chromium/chrome/browser/logo/LogoMediator.java b/chrome/browser/ui/android/logo/java/src/org/chromium/chrome/browser/logo/LogoMediator.java
--- a/chrome/browser/ui/android/logo/java/src/org/chromium/chrome/browser/logo/LogoMediator.java
+++ b/chrome/browser/ui/android/logo/java/src/org/chromium/chrome/browser/logo/LogoMediator.java
@@ -98,6 +98,9 @@ public class LogoMediator implements TemplateUrlServiceObserver {
     private @Nullable Drawable mDefaultGoogleLogoDrawable;
     private boolean mShouldShowLogo;
     private boolean mIsDefaultSearchEngineGoogle;
+    // CromIKO: the built-in CromIKO logo is shown for every search engine; doodles and
+    // search provider logos are never fetched.
+    private static final boolean SHOW_BUILT_IN_LOGO_ONLY = true;
     private boolean mIsLoadPending;
     private boolean mIsDestroyed;
     private @Nullable String mOnLogoClickUrl;
@@ -247,7 +250,7 @@ public class LogoMediator implements TemplateUrlServiceObserver {
 
     /** Returns whether the default Google Logo is shown. */
     boolean isDefaultGoogleLogoShown() {
-        return mIsDefaultSearchEngineGoogle
+        return (mIsDefaultSearchEngineGoogle || SHOW_BUILT_IN_LOGO_ONLY)
                 && mShouldShowLogo
                 && mLogoModel.get(LogoProperties.VISIBILITY)
                 && mLogoModel.get(LogoProperties.LOGO) == null;
@@ -263,6 +266,12 @@ public class LogoMediator implements TemplateUrlServiceObserver {
         // record, don't bother loading the logo image.
         if (mHasLogoLoadedForCurrentSearchEngine || mProfile == null || !mShouldShowLogo) return;
 
+        if (SHOW_BUILT_IN_LOGO_ONLY) {
+            mHasLogoLoadedForCurrentSearchEngine = true;
+            showSearchProviderInitialView();
+            return;
+        }
+
         if (mLogoBridge == null) {
             mLogoBridge = new LogoBridge(mProfile);
         }
@@ -352,6 +361,7 @@ public class LogoMediator implements TemplateUrlServiceObserver {
      */
     @VisibleForTesting
     @Nullable Drawable getDefaultGoogleLogoDrawable() {
+        if (SHOW_BUILT_IN_LOGO_ONLY) return mDefaultGoogleLogoDrawable;
         if (mProfile == null
                 || !TemplateUrlServiceFactory.getForProfile(mProfile)
                         .isDefaultSearchEngineGoogle()) {
diff --git a/components/search_engines/android/java/src/org/chromium/components/search_engines/TemplateUrlService.java b/components/search_engines/android/java/src/org/chromium/components/search_engines/TemplateUrlService.java
--- a/components/search_engines/android/java/src/org/chromium/components/search_engines/TemplateUrlService.java
+++ b/components/search_engines/android/java/src/org/chromium/components/search_engines/TemplateUrlService.java
@@ -197,8 +197,8 @@ public class TemplateUrlService {
      *     surface.
      */
     public boolean doesDefaultSearchEngineHaveLogo() {
-        return TemplateUrlServiceJni.get()
-                .doesDefaultSearchEngineHaveLogo(mNativeTemplateUrlServiceAndroid);
+        // CromIKO: the new tab page always shows the built-in CromIKO logo.
+        return true;
     }
 
     /**
--
